src/RecordSession.cc

```diff
@@ -148,7 +156,7 @@ static bool handle_ptrace_exit_event(RecordTask* t) {
       // that state replay of the SCHED will fail. So detect that state and fix
       // it up.
       if (t->regs().original_syscallno() >= 0 &&
-          t->regs().syscall_result_signed() == -ENOSYS) {
+          looks_like_syscall_entry(t)) {
```

Comment:
I don't really get this. If it's ok for this to always be true on AArch64, why do we need the -ENOSYS check on x86?

Reply:
I'm not really sure. As the comment immediately below indicates, the check is unreliable anyway. The most common reason we hit this path is due to SECCOMP_RET_KILL, though I think we should probably just record the syscall entry event there rather than relying on this. Thinking a bit more along these lines, I'm wondering whether we should just check if the currently instruction is a syscall instruction and if so always putting a sched event right before the syscall instruction. It seems a bit odd here to hallucinate a syscall entry if the registers happen to look like it.

Reply:
Actually, the original_syscallno test doesn't work either on aarch64, because that'll get reset to -1 before we ever get there (the only place it does work is from SECCOMP_RET_KILL), not that we actually ever asked the tracee about it. I think what we need to do here is:
1. Record ip_at_last_recorded_syscall_exit in addition to ticks_at_last_recorded_syscall_exit.
2. Look at the instruction that's currently executing. If it's an svc, we set the syscall entry, otherwise a sched.